src/resources/ble_esp32.cc

```diff
@@ -901,14 +902,12 @@ void
 BleServiceResource::_on_characteristic_discovered(const BleCallbackScope& scope, const struct ble_gatt_error* error, const struct ble_gatt_chr* chr) {
   switch (error->status) {
     case 0: {
-      for (int i = 0; i < 2; i++) {
-        auto ble_characteristic =
-            get_or_create_characteristics_resource(
-                chr->uuid, chr->properties, chr->def_handle,
-                chr->val_handle);
-        if (!ble_characteristic) {
-          set_malloc_error(true);
-        }
+      auto ble_characteristic =
+          get_or_create_characteristics_resource(&scope,
+                                                  chr->uuid, chr->properties, chr->def_handle,
```

Comment:
Isn't the indentation slightly weird here?

Reply:
done.